### PR TITLE
OTPview and BaseInitializableControl

### PR DESCRIPTION
**Pull request by Loupehope**
Add OTPSwiftView and BaseInitializableControl

### CODE REVIEW

File: OTPSwiftView/Sources/Extensions/UITextField/UITextField+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+
+public extension UITextField {
+    var unwrappedText: String {
+        text ?? ""
```
Comment (petropavel13) [inline OTPSwiftView/Sources/Extensions/UITextField/UITextField+Extensions.swift:27]:
```swift
extension Optional where Wrapped == String {
    var orEmpty: Wrapped {
        value ?? .init()
    }
}
```

```swift
textField.text.orEmpty
```

TISwiftUtils?

File: OTPSwiftView/Sources/Helpers/Typealias.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+
+public typealias Spacing = [Int: CGFloat]
+public typealias ValueClosure<T> = ((T) -> Void)
+public typealias VoidClosure = (() -> Void)
```
Comment (petropavel13) [inline OTPSwiftView/Sources/Helpers/Typealias.swift:27]: TISwiftUtils

File: OTPSwiftView/Sources/Extensions/Substring/Substring+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+public extension Substring {
+    var string: String {
+        String(self)
+    }
+}
```
Comment (petropavel13) [inline OTPSwiftView/Sources/Extensions/Substring/Substring+Extensions.swift:29]: TISwiftUtils?

File: OTPSwiftView/Sources/Views/OTPSwiftView/OTPSwiftView.swift [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+// MARK: - Configure textfields
+
+private extension OTPSwiftView {
+    func configure(customSpacing: Spacing?, for stackView: UIStackView) {
+        guard let customSpacing = customSpacing else {
+            return
+        }
+        
+        customSpacing.forEach { [weak self] viewIndex, spacing in
```
Comment (petropavel13) [inline OTPSwiftView/Sources/Views/OTPSwiftView/OTPSwiftView.swift:101]: closure is non-escaping, so weak is not required

File: OTPSwiftView/Sources/Views/OTPTextField/OTPTextField.swift [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        case 0:
+            textField.set(inputText: string)
+            
+            let currentTextField = textField.lastNotEmpty.nextTextField ?? textField.lastNotEmpty
+            currentTextField.becomeFirstResponder()
+            textField.onTextChangedSignal?()
+            
+            return false
+            
+        case 1:
```
Comment (petropavel13) [inline OTPSwiftView/Sources/Views/OTPTextField/OTPTextField.swift:121]: add a comment

File: OTPSwiftView/Sources/Views/OTPTextField/OTPTextField.swift [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        }
+        
+        let isInputEmpty = textField.unwrappedText.isEmpty && string.isEmpty
+        
+        guard isInputEmpty || validationClosure?(string) ?? true else {
+            return false
+        }
+        
+        switch range.length {
+        case 0:
```
Comment (petropavel13) [inline OTPSwiftView/Sources/Views/OTPTextField/OTPTextField.swift:112]: add a comment

File: OTPSwiftView/Sources/Extensions/UITextField/UITextField+Extensions.swift (hunk 1 above)
Comment (krasich74) [reply OTPSwiftView/Sources/Extensions/UITextField/UITextField+Extensions.swift:27]: Можно будет сделать после выноса Utils от нас как таргета)

**Review by krasich74 — APPROVED** [review]

File: TISwiftUtils/Sources/Helpers/Typealias.swift [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+
+public typealias ValueClosure<T> = ((T) -> Void)
+public typealias VoidClosure = (() -> Void)
+public typealias ValidationClosure<T> = ((T) -> Bool)
```
Comment (petropavel13) [inline TISwiftUtils/Sources/Helpers/Typealias.swift:27]: looks like pretty specific closure. please move it to OTP

File: TISwiftUtils/Sources/Helpers/Typealias.swift [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+
+public typealias ValueClosure<T> = ((T) -> Void)
```
Comment (petropavel13) [inline TISwiftUtils/Sources/Helpers/Typealias.swift:25]: LeadKit ParameterClosure naming is better

File: TISwiftUtils/Sources/Helpers/Typealias.swift [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+
```
Comment (petropavel13) [inline TISwiftUtils/Sources/Helpers/Typealias.swift:24]: please, add LeadKit closures

```swift
/// Closure with custom arguments and return value.
public typealias Closure<Input, Output> = (Input) -> Output

/// Closure with no arguments and custom return value.
public typealias ResultClosure<Output> = () -> Output

/// Closure that takes custom arguments and returns Void.
public typealias ParameterClosure<Input> = Closure<Input, Void>

// MARK: Throwable versions

/// Closure with custom arguments and return value, may throw an error.
public typealias ThrowableClosure<Input, Output> = (Input) throws -> Output

/// Closure with no arguments and custom return value, may throw an error.
public typealias ThrowableResultClosure<Output> = () throws -> Output

/// Closure that takes custom arguments and returns Void, may throw an error.
public typealias ThrowableParameterClosure<Input> = ThrowableClosure<Input, Void>

// MARK: Concrete closures

/// Closure that takes no arguments and returns Void.
public typealias VoidClosure = ResultClosure<Void>

/// Closure that takes no arguments, may throw an error and returns Void.
public typealias ThrowableVoidClosure = () throws -> Void
```

**Review by petropavel13 — APPROVED** [review]

**Pull request merged**